extensions/amp-story/1.0/progress-bar.js

```diff
@@ -119,10 +180,191 @@ export class ProgressBar {
       true /** callToInitialize */
     );
 
+    this.storeService_.subscribe(
+      StateProperty.RTL_STATE,
+      rtlState => {
+        this.onRtlStateUpdate_(rtlState);
+      },
+      true /** callToInitialize */
+    );
+
+    Services.viewportForDoc(this.ampdoc_).onResize(
+      debounce(this.win_, () => this.onResize_(), 300)
+    );
+
+    this.resources_.measureElement(() => {
+      this.barWidthPx_ =
+        this.storyEl_
+          .querySelector('amp-story-page')
+          ./*OK*/ getBoundingClientRect().width - PROGRESS_BAR_PADDING_PX;
+    });
+
     this.isBuilt_ = true;
     return this.getRoot();
   }
 
+  /**
+   * Reacts to story replay.
+   * @private
+   */
+  replay_() {
+    this.firstExpandedSegmentIndex_ = 0;
+    this.render_(false /** shouldAnimate */);
+  }
+
+  /**
+   * Renders the segments by setting their corresponding scaleX and translate.
+   * @param {boolean} shouldAnimate
+   * @private
+   */
+  render_(shouldAnimate = true) {
+    const segmentWidth = this.getSegmentWidth_();
+    let translateX =
+      -(this.firstExpandedSegmentIndex_ - this.getPrevEllipsisCount_()) *
+      (ELLIPSE_WIDTH_PX + SEGMENTS_MARGIN_PX);
+
+    this.resources_
+      .mutateElement(this.root_, () => {
+        for (let index = 0; index < this.segmentCount_; index++) {
+          const width =
+            index >= this.firstExpandedSegmentIndex_ &&
+            index < this.firstExpandedSegmentIndex_ + MAX_SEGMENTS
+              ? segmentWidth
+              : ELLIPSE_WIDTH_PX;
+          this.transform_(this.segments_[index], translateX, width);
+          translateX += width + SEGMENTS_MARGIN_PX;
+        }
+      })
+      .then(() => {
+        this.root_.classList.toggle(
+          'i-amphtml-animate-progress',
+          shouldAnimate
+        );
```

Comment:
Does this work? I'm a bit confused since you're toggling the class after setting the transform properties.